### DISCUSSION TITLE
change SubViewportContainer stretch_shrink's type from `int` to `float`

### DISCUSSION THREAD

**Discussion by jinyangcruise** [godotengine/godot-proposals#6921] [GUI]
<img width="609" alt="image" src="https://github.com/godotengine/godot-proposals/assets/42859362/16f8129e-d35e-4dc5-ab3c-2bdd378c6da2">

Anyone know the reason `stretch_shrink` must be an `int`? I think a `float` can make it more useful.

I find it's hard to change the scale of a scene which is made by a bounch of Control node. The scene is a child of a CenterContainer, so it's position and size is determined only by it's parent the CenterContainer. I finally find a work around: add the scene to a SubViewport then add the SubViewport to a SubViewportContainer, then add the SubViewportContainer to the CenterContainer. Then I change the stretch_shrink of the SubViewportContainer to make the scene bigger. 

stretch_shrink's type is int and must be greater than 1, this situation possibly limits it's ability. I don't know whether I used subviewport in a correct way.

<img width="227" alt="image" src="https://github.com/godotengine/godot-proposals/assets/42859362/4245c788-7033-439a-8d96-821b058ee331">

**Comment by AThousandShips** [comment]
What would a value smaller than one do? This is to speed up rendering by downscaling the viewport

It being an int makes sense or you'll have bad scaling factors very easily 

Also it has to be greater than *or equal* to 1

> I don't know whether I used subviewport in a correct way.

I suspect that you might not

**Reply by jinyangcruise** [reply]
Yes, you are right.

**Comment by Calinou** [comment]
If you want your project to scale to various screen resolutions, use Godot's built-in support for [multiple resolutions](https://docs.godotengine.org/en/stable/tutorials/rendering/multiple_resolutions.html) (stretch modes) instead of implementing scaling manually.

**Comment by jinyangcruise** [comment]
I should not use a viewport to scale my `Wheel` scene. The direct way is to add a control that has a min_size to contain my `Wheel` scene. Then my `Wheel` scene can be scaled to another value.
